File: python/Node.py
class Node:
    def __init__(self, completeWordFlag):
        self.__completeWordFlag = completeWordFlag
        self.__children = [None] * 26
    
    def getChildren(self):
        return self.__children
    
    def setChild(self, character: chr, completeWordFlag: bool):
        child = Node(completeWordFlag)
        self.__children[ord(character)-97] = child
        return child

    def isCompleteWord(self):
        return self.__completeWordFlag

File: node/Node.js
class Node {
  #completeWordFlag;
  #children;
  
  constructor(completeWordFlag) {
    this.#completeWordFlag = completeWordFlag;
    this.#children = new Array(26);
  }
  
  getChildren(){
    return this.#children;
  }
  
  setChild(character, completeWordFlag){
    const child = new Node(completeWordFlag);
    this.#children[character.charCodeAt(0)-97] = child;

    return child;
  }
  
  isCompleteWord(){
    return this.#completeWordFlag;
  }
  
  setCompleteWord(completeWordFlag){
    this.#completeWordFlag = completeWordFlag;
  }
}

module.exports = Node;
